docs/workflow/developer-workflows.md

```diff
@@ -0,0 +1,441 @@
+that can be supplied in a template:
+
+```
+$ helm list-params helm:example.com/foo/bar
+Params:
+- bgcolor: The background color for the home page (hex code or HTML colors)
+- title: The title of the app
+```
+
+- The client sends the request to the API server
+- The API server fetches the chart, analyzes it, and returns the list of
+  parameters.
+
+#### Generate the params for me:
+
+In this operation, helm generates a parameter values file for the user.
+
+```
+$ helm gen-params helm:example.com/foo/bar
+Created: values.yaml
+$ edit values.yaml
+```
+
+- The client sends the request to the server
+- The server returns a stub file
+- The client writes the file to disk
+
+#### Deploy with params:
+
+In this operation, the user deploys a chart with an associated values
+file.
+
+```
+$ helm deploy helm:example.com/foo/bar values.yaml
+Created taco-tuesday
+```
+
+- The client sends a request with the name of the chart and the values
+  file.
+- The API server generates a name (`taco-tuesday`)
+- The API server fetches the request chart, sends the chart and values
+  through the template resolution phase, and deploys the results.
+
+If we allow the user to pass in a name (overriding the generated name),
+then the server must first guarantee that this name is unique.
+
+#### Get the info about named deployment.
+
+A deployment, as we have seen, is a named instance of a chart.
+Operations that operate on these instances use the name to refer to the
+instance.
+
+```
+$ helm status taco-tuesday
+OK
+Located at: 10.10.10.77:8080
+```
+
+- The client sends the API server a request for the status of
+  `taco-tuesday`.
+- The API server looks up pertinent data and returns it.
+- The client displays the data.
+
+#### Edit and redeploy:
+
+Redeployment is taking an existing _instance_ and changing its template
+values, and then re-deploying it.
+
+```
+$ edit values.yaml
+$ helm redeploy taco-tuesday values.yaml
+Redeployed taco-tuesday
+```
+
+- The client sends the instance name and the new values to the server
+- The server coalesces the values into the existing instance and then
+  restarts.
+
+#### Uninstall:
+
+```
+$ helm uninstall taco-tuesday
+Destroyed taco-tuesday
+```
+
+- The client sends the DELETE instance name command
+- The API server destroys the resource
+
+### Power User Features
+
+Users familiar with the system may desire additional tools.
+
+#### Name a deploy
+
+```
+$ helm deploy -name skinny-pigeon example.com/foo/bar
+```
+
+This follows the deployment process above. The server _must_ ensure that
+the name is unique.
+
+#### Get values for an app:
+```
+$ helm get-values taco-tuesday
+Stored in values.yaml
+$ helm redeploy taco-tuesday values.yaml
+```
+
+- The client sends the instance name to the values endpoint
+- The server returns the values file used to generate the instance.
+- The client writes this to a file (or, perhaps, to STDOUT)
+
+#### Get fully generated manifest files
```

Comment:
Currently, we maintain a history of manifests. We should probably provide a flag to let the user download only the latest, which represents the current deployment, or download the full history. Not sure which should be the default. Maybe just the latest, with a flag to get the history.